addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/handler/KNXGenericThingHandler.java

```diff
@@ -0,0 +1,1124 @@
+
+        Configuration channelConfiguration = getThing().getChannel(channelUID.getId()).getConfiguration();
+        Boolean mustRead = (Boolean) channelConfiguration.get(READ);
+        BigDecimal readInterval = (BigDecimal) channelConfiguration.get(INTERVAL);
+
+        KNXChannelSelector selector = KNXChannelSelector.getValueSelectorFromChannelTypeId(
+                getThing().getChannel(channelUID.getId()).getChannelTypeUID().getId());
+
+        if (selector != null) {
+            try {
+                for (GroupAddress address : knxChannelSelectorProxy.getReadAddresses(selector, channelConfiguration,
+                        null)) {
+                    if (mustRead || (readInterval != null && readInterval.intValue() > 0)) {
+                        scheduleReadJob(address,
+                                knxChannelSelectorProxy.getDPT(address, selector, channelConfiguration, null), true,
+                                BigDecimal.ZERO);
+                    }
+                }
+            } catch (KNXFormatException e) {
+                logger.error("An exception occurred while scheduling a read job : '{}'", e.getMessage());
+            }
+        }
+    }
+
+    private void scheduleReadJob(GroupAddress groupAddress, String dpt, boolean mustRead, BigDecimal readInterval) {
+
+        ReadRunnable readRunnable = null;
+        if (mustRead || (readInterval != null && readInterval.intValue() > 0)) {
+            readRunnable = new ReadRunnable(groupAddress, dpt);
+        }
+
+        if (readRunnable != null) {
+            if (readInterval != null && readInterval.intValue() > 0) {
+                readFutures.add(
+                        scheduler.scheduleWithFixedDelay(readRunnable, 0, readInterval.intValue(), TimeUnit.SECONDS));
+            } else if (mustRead) {
+                scheduler.schedule(readRunnable, 0, TimeUnit.SECONDS);
+            }
+        }
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        super.bridgeStatusChanged(bridgeStatusInfo);
+
+        if (getThing().getStatus() == ThingStatus.ONLINE
+                && getThing().getStatusInfo().getStatusDetail() == ThingStatusDetail.NONE) {
+            ((KNXBridgeBaseThingHandler) getBridge().getHandler()).registerGroupAddressListener(this);
+
+            for (Channel channel : getThing().getChannels()) {
+                Configuration channelConfiguration = channel.getConfiguration();
+                Boolean mustRead = (Boolean) channelConfiguration.get(READ);
+                BigDecimal readInterval = (BigDecimal) channelConfiguration.get(INTERVAL);
+
+                KNXChannelSelector selector = KNXChannelSelector
+                        .getValueSelectorFromChannelTypeId(channel.getChannelTypeUID().getId());
+
+                if (selector != null) {
+                    try {
+                        for (GroupAddress address : knxChannelSelectorProxy.getReadAddresses(selector,
```

Comment:
Although not yet used in this binding, the proxy is a mechanism to pass on variables to the selector, but also a placeholder to store important information for the selectors, like a state, history and so forth. Such information can not be stored in the selectors, as it is an Enum of course. Consider this be for further use, as we might need this in more complicated scenario's as we go along and add channels. For a working example, pls see the Tesla binding.